### DISCUSSION TITLE
require() of ES Module is not supported

### DISCUSSION THREAD

**Discussion by DoneDeal0** [node-fetch/node-fetch#1544] [Q&A]
Hi,

I know v3 only supports ESM. I've read several threads, including the official one pinned at the top of the repo, but I just can't make it work. I use node-fetch from the cli in a frontend project to fetch remote data and update local files (translations).

The cli command is:
```json
 "translate": "yarn babel-node --root-mode upward --ignore=node_modules tasks/translate",
```

My original version is (works fine with v2.6.0, crashes with v3):

```js
import fetch from "node-fetch";

const saveBundle = async (url, path) => {
  try {
    const response = await fetch(url);
    const buffer = await response.buffer();
    const zip = new AdmZip(buffer);
    zip.extractAllTo(path, true);
  } catch (error) {
    throw new Error(error.message);
  }
};
```

So I've tried to use a dynamic import instead (as suggested in your thread):
```js
const _importDynamic = new Function("modulePath", "return import(modulePath)");

async function fetch(...args) {
  const { default: fetch } = await _importDynamic("node-fetch");
  return fetch(...args);
}

const saveBundle = async (url, path) => {
  try {
    const response = await fetch(url);
    const buffer = await response.buffer();
    const zip = new AdmZip(buffer);
    zip.extractAllTo(path, true);
  } catch (error) {
    throw new Error(error.message);
  }
};
```

How to fix this? Thank you!

**Comment by angelhdzmultimedia** [comment]
Same issue. I am making a simple RESTful API CRUD with TypeScript and Express (backend) and Vite and Vue (frontend). Everything works with ESM, except node-dev...
and ironic when v3 says it's made only with ESM... 🤔

**Reply by jimmywarting** [reply]
update to v18 and use the built in fetch: https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#fetch
